src/main/java/net/vulkanmod/vulkan/memory/AutoIndexBuffer.java

```diff
@@ -44,89 +48,78 @@ private void createIndexBuffer(int vertexCount) {
     }
 
     public void checkCapacity(int vertexCount) {
-        if(vertexCount > this.vertexCount) {
-            int newVertexCount = this.vertexCount * 2;
-            System.out.println("Reallocating AutoIndexBuffer from " + this.vertexCount + " to " + newVertexCount);
-
-            //TODO: free old
-            //Can't know when VBO will stop using it
-            indexBuffer.freeBuffer();
-            createIndexBuffer(newVertexCount);
+        if (vertexCount <= this.vertexCount) {
+            return;
         }
-    }
 
-    public static ByteBuffer genQuadIdxs(int vertexCount) {
-        //short[] idxs = {0, 1, 2, 0, 2, 3};
+        System.out.println("Reallocating AutoIndexBuffer from " + this.vertexCount + " to " + vertexCount);
 
-        int indexCount = vertexCount * 3 / 2;
-        ByteBuffer buffer = MemoryUtil.memAlloc(indexCount * Short.BYTES);
-        ShortBuffer idxs = buffer.asShortBuffer();
-        //short[] idxs = new short[indexCount];
+        // TODO: free old
+        // Can't know when VBO will stop using it
+        indexBuffer.freeBuffer();
+        createIndexBuffer(vertexCount);
+    }
 
-        int j = 0;
-        for(int i = 0; i < vertexCount; i += 4) {
+    public static ByteBuffer genQuadIndices(int vertexCount) {
+        ByteBuffer buffer = MemoryUtil.memCalloc(vertexCount / 4 * 6 * Short.BYTES);
+        ShortBuffer indices = buffer.asShortBuffer();
 
-            idxs.put(j, (short) i);
-            idxs.put(j + 1, (short) (i + 1));
-            idxs.put(j + 2, (short) (i + 2));
-            idxs.put(j + 3, (short) (i));
-            idxs.put(j + 4, (short) (i + 2));
-            idxs.put(j + 5, (short) (i + 3));
+        int j = 0;
+        for (int i = 0; i < vertexCount; i += 4) {
+            indices.put(j, (short) i);
+            indices.put(j + 1, (short) (i + 1));
+            indices.put(j + 2, (short) (i + 2));
+            indices.put(j + 3, (short) (i + 2));
+            indices.put(j + 4, (short) (i + 3));
+            indices.put(j + 5, (short) i);
 
             j += 6;
         }
 
         return buffer;
-        //this.type.copyIndexBuffer(this, bufferSize, idxs);
     }
 
-    public static ByteBuffer genTriangleFanIdxs(int vertexCount) {
```

Comment:
The Skybox gradient doesn't render correctly in this PR
Issues like this are usually due to incorrect index ordering, so not a big deal to fix most likely

UpStream
![b](https://github.com/xCollateral/VulkanMod/assets/125277899/7f736b88-8bd5-421d-86ad-5cb32184c1c0)

This PR
![a](https://github.com/xCollateral/VulkanMod/assets/125277899/90422950-2624-4608-a708-89687b0b5c8d)



Think the Skybox uses TriangleFan topology, which is why I put this comment here, but could easily be wrong about that though.

Reply:
<img width="1552" alt="Screenshot 2024-05-09 at 23 04 36" src="https://github.com/xCollateral/VulkanMod/assets/30368284/09c2e75c-07c8-40a2-9b86-d1da4f9ac5d8">
This looks fine? @thr3343

Reply:
Afaict this is potentially a driver/hardware-specific issue, as I can reproduce it without fail on my Nvidia GPU

This not the fault of the PR: Hardware specific bugs like this are the absolute worst to debug as they are impossible to reproduce without the right GPU.

There are some Nvidia users in the Discord, so sometime later will ask them to test this PR, to confirm if it's indeed Nvidia-specific or if it just an issue on my System.

![x](https://github.com/xCollateral/VulkanMod/assets/125277899/120fe46c-85e9-411a-a625-0a4a4d0636ae)

Reply:
It is not. I just made a typo and will commit the fix when possible.